pkg/services/sqlstore/team.go

```diff
@@ -334,27 +320,88 @@ func getTeamMember(sess *DBSession, orgId int64, teamId int64, userId int64) (mo
 // UpdateTeamMember updates a team member
 func (ss *SQLStore) UpdateTeamMember(ctx context.Context, cmd *models.UpdateTeamMemberCommand) error {
 	return inTransaction(func(sess *DBSession) error {
-		member, err := getTeamMember(sess, cmd.OrgId, cmd.TeamId, cmd.UserId)
+		return updateTeamMember(sess, cmd.OrgId, cmd.TeamId, cmd.UserId, cmd.Permission)
+	})
+}
+
+func (ss *SQLStore) IsTeamMember(orgId int64, teamId int64, userId int64) (bool, error) {
+	var isMember bool
+
+	err := ss.WithTransactionalDbSession(context.Background(), func(sess *DBSession) error {
+		var err error
+		isMember, err = isTeamMember(sess, orgId, teamId, userId)
+		return err
+	})
+
+	return isMember, err
+}
+
+func isTeamMember(sess *DBSession, orgId int64, teamId int64, userId int64) (bool, error) {
+	if res, err := sess.Query("SELECT 1 FROM team_member WHERE org_id=? and team_id=? and user_id=?", orgId, teamId, userId); err != nil {
+		return false, err
+	} else if len(res) != 1 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// SaveTeamMember adds user to a team or updates user permissions in a team
+func (ss *SQLStore) SaveTeamMember(userID, orgID, teamID int64, isExternal bool, permission models.PermissionType) error {
```

Comment:
It's a bit hard to distinguish SaveTeamMember from UpdateTeamMember. Is there a better name for this method?

Reply:
I agree, it's not the best name. What about `AddOrUpdateTeamMember`? It's long, but at least it's clear what it does.